cli/src/commands/init/index.ts

```diff
@@ -1,17 +1,163 @@
+import prompts from 'prompts';
+import which from 'which';
 import { BaseCommand } from '../../base.js';
+import { readAPIKey } from '../../key.js';
 
 export default class Init extends BaseCommand {
-  static description = 'Initialize a database';
+  static description = 'Configure your working directory to work with a Xata database';
 
   static examples = [];
 
-  static flags = {};
+  static flags = {
+    databaseURL: this.databaseURLFlag
+  };
 
-  static args = [];
+  static args = [
+    // TODO: add an arg for initial schema
+  ];
+
+  packageManager: 'npm' | 'yarn' | undefined;
 
   async run(): Promise<void> {
-    // const { args, flags } = await this.parse(Init);
+    const { flags } = await this.parse(Init);
+
+    if (this.projectConfigLocation) {
+      // TODO: have a --force option
+      this.error(`Project already configured at ${this.projectConfigLocation}`);
+    }
+
+    const databaseURL = await this.getDatabaseURL(flags.databaseURL, true);
+
+    this.projectConfig = { databaseURL };
+
+    await this.installSDK();
+
+    await this.editSchema();
+
+    await this.runCodegen();
+
+    await this.writeConfig();
+
+    await this.writeEnvFile();
+
+    this.log('Done. You are all set!');
+  }
+
+  async installSDK() {
+    const result = await prompts({
+      type: 'confirm',
+      name: 'confirm',
+      message: 'Do you want to install the TypeScript/JavaScript SDK?',
+      initial: true
+    });
+    if (!result.confirm) return;
+
+    await this.installPackage('@xata.io/codegen');
+
+    await this.configureCodegen();
+  }
+
+  async configureCodegen() {
+    const result = await prompts({
+      type: 'confirm',
+      name: 'confirm',
+      message: 'Do you want to use the TypeScript/JavaScript code generator?',
+      initial: true
+    });
+    if (!result.confirm) return;
+
+    const config = await prompts([
+      {
+        type: 'text',
+        name: 'output',
+        message: 'Choose where the output file for the code generator',
+        initial: 'src/xata.ts'
+      },
+      {
+        type: 'confirm',
+        name: 'declarations',
+        message: 'Do you want to generate the TypeScript declarations?',
+        initial: (prev) => !prev.endsWith('.ts')
+      }
+    ]);
+
+    this.projectConfig = this.projectConfig || {};
+    this.projectConfig.codegen = config;
+  }
+
+  async editSchema() {
+    // TODO
+  }
+
+  async runCodegen() {
+    // TODO
+  }
+
+  async getPackageManager() {
+    if (this.packageManager) return this.packageManager;
+
+    const packageManager = (await this.access('yarn.lock')) ? 'yarn' : 'npm';
+    if (!which.sync(packageManager, { nothrow: true })) {
+      this.error(
+        `Looks like ${packageManager} is not installed or is not in the PATH. This made impossible to install the code generator`
+      );
+    }
+    this.packageManager = packageManager;
+    return packageManager;
+  }
+
+  async access(path: string) {
+    try {
+      await access(path);
```

Comment:
What happens if I call xata from a subdir in my project? This only checks on the current folder, right?

Reply:
In this case you are initalizing the project, so it is supposed you want to initialize it in the current dir. Like when you do `npm init`, it creates a `package.json` file in the current dir.